spec/helpers/tag_sets_helper_spec.rb

```diff
@@ -0,0 +1,211 @@
+      it 'should mention characters and relationships' do
+        @limit[:fandom] = 0
+        expect(helper.nomination_notes(@limit))
+          .to match('You can nominate up to 3 characters and 3 relationships.')
+      end
+    end
+
+    context 'for nominations allowing characters, NO relationships,
+        NO fandoms' do
+      it 'should mention only characters' do
+        @limit[:fandom] = 0
+        @limit[:relationship] = 0
+        @limit[:freeform] = 0
+        expect(helper.nomination_notes(@limit))
+          .to eq('You can nominate up to 3 characters.')
+      end
+    end
+
+    context 'for nominations allowing relationships, NO characters,
+        NO fandoms' do
+      it 'should mention only relationships' do
+        @limit[:fandom] = 0
+        @limit[:character] = 0
+        @limit[:freeform] = 0
+        expect(helper.nomination_notes(@limit))
+          .to eq('You can nominate up to 3 relationships.')
+      end
+    end
+  end
+
+  describe 'nomination_status' do
+    before(:each) do
+      @nomination = FactoryGirl.create(:tag_nomination)
+    end
+
+    context 'for valid nominations' do
+      it 'should show basic information' do
+        expect(helper.nomination_status(@fake_nomination))
+          .to include('unreviewed').and include('?!')
+          .and include('This nomination has not been reviewed yet.')
+      end
+    end
+
+    context 'for approved nominations' do
+      it 'should show correct class information' do
+        @nomination.approved = true
+        expect(helper.nomination_status(@nomination)).to include('approved')
+          .and include('This nomination has been approved!')
+          .and include('&#10004;')
+      end
+    end
+
+    context 'for rejected nominations' do
+      it 'should show correct class information' do
+        @nomination.rejected = true
+        @nomination.approved = false
+        expect(helper.nomination_status(@nomination)).to include('rejected')
+          .and include('This nomination was rejected')
+          .and include('&#10006;')
+      end
+    end
+
+    context 'for unreviewed nominations' do
+      it 'should show correct class information' do
+        @nomination.owned_tag_set.nominated = true
+        @nomination.approved = ''
+        @nomination.rejected = ''
+        @tag_set = FactoryGirl.create(:owned_tag_set)
+        expect(helper.nomination_status(@nomination)).to include('unreviewed')
+          .and include('has not been reviewed yet and can still be changed.')
+          .and include('?!')
+      end
+    end
+  end
+
+  describe 'nomination_tag_information' do
+    before(:each) do
+      @nominated_tag = FactoryGirl.create(:tag_nomination)
+    end
+
+    context 'for valid information' do
+      it 'should show basic information' do
+        @nominated_tag.canonical = false
+        @nominated_tag.exists = false
+        expect(helper.nomination_tag_information(@nominated_tag))
+          .to include('nonexistent')
```

Comment:
I've been bitten on the ass by this a couple times (because apparently it does sometimes break things /o\), so if you have a minute, it might be a good idea to follow Hound's advice here.